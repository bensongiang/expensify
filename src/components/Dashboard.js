import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary';

const Dashboard = () => (
  <div>
    <h1>Dashboard</h1>
    <ExpenseListFilters />
    <ExpensesSummary />
    <ExpenseList />
  </div> 
);
export default Dashboard; 