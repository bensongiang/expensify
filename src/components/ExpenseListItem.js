import React from 'react';
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expenses';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

export const ExpenseListItem = ({ dispatch, id, description, amount, createdAt }) => (
  <div className="expense-item">
    <p><Link to={{ pathname: `/edit/${id}` }}>{description}</Link></p>
    <p>{numeral(amount / 100).format('$0,0.00')}</p>
    <p>{moment(createdAt).format('MMMM Do, YYYY')}</p>
    <button 
      className="button-link"
      onClick={() =>
        {
          dispatch(removeExpense({ id }));
        }}
    >Remove</button>
  </div>
);

export default connect()(ExpenseListItem);