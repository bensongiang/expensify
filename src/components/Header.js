import React from 'react';
import { Link, NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
  <header className="header">
    <Link to="/dashboard" className="header-logo"><h1>Expensify</h1></Link>
    <div className="navigation">
      <NavLink to="/dashboard">Dashboard</NavLink>
      <NavLink to="/add">Add</NavLink>
      <button onClick={startLogout}>Logout</button>
    </div>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);