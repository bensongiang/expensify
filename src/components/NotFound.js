import React from 'react';
import { Link, NavLink } from 'react-router-dom';

const NotFound = () => (
  <div>
    <h1>404! Whoa, this page does not exist.</h1>
    <Link to="/">Take me home!</Link>
  </div> 
);

export default NotFound;