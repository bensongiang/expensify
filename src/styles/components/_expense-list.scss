.expense-list {
  display: flex;
  flex-direction: column;
}

.expense-item {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  border-bottom: 1px solid $brand-color-light-grey;
}