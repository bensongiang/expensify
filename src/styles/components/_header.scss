.header {
  background: $brand-color-dark-blue;
  padding: $m-size;
  display: flex;
  justify-content: space-between;
  align-items: center;
  flex-direction: column;
  text-decoration: none;
  color: white;

  .header-logo {
    flex-grow: 1;
  }

  a {
    color: white;
    text-decoration: none;
  }

  .active {
    color: $brand-color-coral;
  }
}

.navigation {
  display: flex;
  justify-content: space-around;
  align-items: center;
  width: 100%;

  
}

@media (min-width: $desktop-breakpoint) {
  .header {
    flex-direction: row;
  }

  .navigation {
    width: auto;

    a {
      padding-left: $xxl-size;
    }
  }
}