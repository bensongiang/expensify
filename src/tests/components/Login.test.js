import React from 'react';
import { shallow } from 'enzyme';
import { Login } from '../../components/Login';

test('Should render Login correctly', () => {
  const wrapper = shallow(<Login startLogin={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});

test('should call startLogin on button click', () => {
  const onSubmitSpy = jest.fn();
  const wrapper = shallow(<Login startLogin={onSubmitSpy} />);
  wrapper.find('button').at(0).simulate('click');

  expect(onSubmitSpy).toHaveBeenCalled();
});