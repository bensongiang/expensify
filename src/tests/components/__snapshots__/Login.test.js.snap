// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Login correctly 1`] = `
<div>
  <h1>
    Expensify
  </h1>
  <button
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
